<?xml version="1.0"?>
<robot xmlns:xacro="http://www.ros.org/wiki/xacro">

  <xacro:macro name="prarob_manipulator_ros2_control" params="name">

    <ros2_control name="${name}" type="system">
      <hardware>
        <plugin>dynamixel_hardware/DynamixelHardware</plugin>
        <param name="usb_port">/dev/ttyUSB0</param>
        <param name="baud_rate">1000000</param>
        <!-- <param name="use_dummy">true</param> -->
      </hardware>
      <joint name="joint1">
        <param name="id">11</param>
        <command_interface name="position"/>
        <command_interface name="velocity"/>
        <state_interface name="position"/>
        <state_interface name="velocity"/>
        <state_interface name="effort"/>
        <param name="Position P Gain">640</param> <!-- Proportional gain for position control -->
        <param name="Position I Gain">0</param> <!-- Integral gain for position control -->
        <param name="Position D Gain">3600</param> <!-- Derivative gain for position control -->
      </joint>
      <joint name="joint2">
        <param name="id">12</param>
        <command_interface name="position"/>
        <command_interface name="velocity"/>
        <state_interface name="position"/>
        <state_interface name="velocity"/>
        <state_interface name="effort"/>
        <param name="Position P Gain">640</param> <!-- Proportional gain for position control -->
        <param name="Position I Gain">0</param> <!-- Integral gain for position control -->
        <param name="Position D Gain">3600</param> <!-- Derivative gain for position control -->
      </joint>
      <joint name="joint3">
        <param name="id">13</param>
        <command_interface name="position"/>
        <command_interface name="velocity"/>
        <state_interface name="position"/>
        <state_interface name="velocity"/>
        <state_interface name="effort"/>
        <param name="Position P Gain">640</param> <!-- Proportional gain for position control -->
        <param name="Position I Gain">0</param> <!-- Integral gain for position control -->
        <param name="Position D Gain">3600</param> <!-- Derivative gain for position control -->
      </joint>
    </ros2_control>

  </xacro:macro>

</robot>
